src/index.ts

```diff
@@ -81,22 +86,136 @@ export class Client {
       headers
     };
 
-    return fetchImpl(url, request).then(response => {
-      return response.json().then(
-        (body: SubmissionBody): SubmissionResponse => {
-          return { body, response };
+    if (this.stripePromise && opts.handlePayment) {
+      // Get Stripe payload
+      const payload = await opts.handlePayment();
```

Comment:
Minor nit: I'm a little confused why we call this handlePayment. Doesn't it just create a paymentMethod so should be called createPaymentMethod or something?

Reply:
Sure I can rename it